### ISSUE TITLE
Terminal filled with ^[[D^[[D^[[D^[[D when alt-clicking a link

### ISSUE THREAD

**Issue by notpushkin** [microsoft/vscode#101136]
- VSCode Version: 1.46.1
- OS Version: Ubuntu 20.04 LTS x86_64
- shell: bash

When alt-clicking in terminal (e. g. to open a file or URL), it gets filled up with lots of `^[[D^[[D^[[D^[[D` or `^[[C^[[C^[[C^[[C`:

![image](https://user-images.githubusercontent.com/1298948/85865933-51fefc80-b7cf-11ea-94e0-4a84aba63feb.png)

Steps to Reproduce:

1. Open integrated terminal
2. Run `read` (or, for a more practical example, `yarn create react-app . && yarn start` in a temp dir)
3. Alt-click anywhere 

Does this issue occur when all extensions are disabled?: Yes

**Comment by davatron5000** [comment]
Having the same issue in WSL with Ubuntu 18.04 when alt+clicking any link.

```
Copied 111 files / Wrote 420 files in 3.46 seconds (8.2ms each, v0.11.0)
Watching…
[Browsersync] Access URLs:
 --------------------------------------
       Local: http://localhost:8080
    External: http://192.168.1.205:8080
 --------------------------------------
          UI: http://localhost:3001
 UI External: http://localhost:3001
 --------------------------------------
[Browsersync] Serving files from: _site
^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D```

**Comment by Tyriar** [comment]
Looks like this is an issue where the alt+click feature (move cursor when clicking) is conflicting with links, only when `"editor.multiCursorModifier": "ctrlCmd"` is used.

**Comment by securedirective** [comment]
This issue also occurs on MacOS, only with Opt+Click instead of Alt+Click.

I thought at first that this was related to the Opt+Click on a detected link in the terminal (since that is an Alt+Click on Windows/Linux). But this truly does occur when Opt+Clicking **anywhere** in the terminal while a `read` command is waiting for input.

@Tyriar I can confirm that this occurs on my MacOS regardless of the value of `"editor.multiCursorModifier"` setting.

**Comment by yulinfeng000** [comment]
set `"terminal.integrated.macOptionIsMeta": true`  in setting.json , now the way you click url is cmd + click, and it wont happen again

**Comment by securedirective** [comment]
@yulinfeng000, nope, that only slightly changes the junk that is dumped into the Terminal, but it doesn't get rid of it...

With `"terminal.integrated.macOptionIsMeta": false` (default):
```
11:20:12 0 user@laptop ~ > read -p "Type something: " testvar
Type something: ^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^C
```

With `"terminal.integrated.macOptionIsMeta": true`:
```
11:20:35 0 user@laptop ~ > read -p "Type something: " testvar
Type something: ^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^C
```

**Comment by yulinfeng000** [comment]
> @yulinfeng000, nope, that only slightly changes the junk that is dumped into the Terminal, but it doesn't get rid of it...
> 
> With `"terminal.integrated.macOptionIsMeta": false` (default):
> 
> ```
> 11:20:12 0 user@laptop ~ > read -p "Type something: " testvar
> Type something: ^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^[[C^C
> ```
> 
> With `"terminal.integrated.macOptionIsMeta": true`:
> 
> ```
> 11:20:35 0 user@laptop ~ > read -p "Type something: " testvar
> Type something: ^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^[[D^C
> ```

emmm, you say is right, i just change a way to open url in vscode terminal to avoid echo ^[[D  but not fix it , this also occurs in terminal.app, item.app if you use option + click.

by the way,my final setting as follows
```
  "editor.multiCursorModifier": "alt",
 "terminal.integrated.macOptionIsMeta": true
```
(if  `"editor.multiCursorModifier": "ctrlcmd"` , vscode  still let you click url with option + click )
![image](https://user-images.githubusercontent.com/44149505/90269207-d6decc00-de8a-11ea-8cda-86a7167db05c.png)

**Comment by securedirective** [comment]
Oh! That's a great observation, @yulinfeng000.... Alt+Click fills the terminal with junk even in the standard MacOS Terminal app and also in iTerm2.

Now, I am aware that @notpushkin is originally reporting this for Ubuntu. I checked my Linux laptop (Fedora 31), and Alt+Click in the standard system terminal emulator **does nothing**. It only occurs in VSCode's terminal.

But this still makes me wonder if all of this is system-level issues, or perhaps terminal "emulator" issues, not really a VSCode issue. Hard to know. But the Microsoft devs would know.

**Comment by securedirective** [comment]
As for your workaround, @yulinfeng000, thank you so much for that!!!  I still don't see any change in VSCode's behavior when using `"terminal.integrated.macOptionIsMeta": true`, but changing `"editor.multiCursorModifier"` definitely avoids the core issue.

I hope you don't mind me describing in more detail...

* According to [VSCode documentation for editor.multiCursorModifier](https://code.visualstudio.com/docs/editor/codebasics#_multicursor-modifier), "The Go To Definition and Open Link gestures will also respect this setting and adapt such that they do not conflict. For example, when the setting is ctrlCmd, multiple cursors can be added with Ctrl/Cmd+Click, and opening links or going to definition can be invoked with Alt+Click."
* You probably did what I did when originally installing VSCode... I'm using to using Ctrl+Click in Linux and Cmd+Click in MacOS to multi-select files in Finder, items in a list, etc. So I set `"editor.multiCursorModifier"` to be `ctrlCmd`. However, this means that attempting to click on a link in the terminal (for example, an error message referencing the offending line in your source code) requires the use of the **other shortcut** (Alt + Click).
* Since it seems that Linux and MacOS have various strange behaviors when Alt + Clicking in a terminal, you are suggesting we switch back to the default `"editor.multiCursorModifier": "alt"`. Now we'll use Ctrl+Click / Cmd+Click to follow links in the Terminal, avoiding all that ^[[D^[[D^[[D^[[D nonsense that sometimes causes the terminal to hang.
* For multi-selecting in the editor window, we'll have to get used to using Alt + Click for that, but in my opinion, that is better than having terminal hanging issues unexpectedly.

**Comment by securedirective** [comment]
So to clarify what I originally posted to @Tyriar... Alt+Click will still cause junk to fill the screen, regardless of the value of "editor.multiCursorModifier" setting. However, the above workaround means we don't need the Alt+Click anymore.

**Comment by Tyriar** [comment]
> Alt+Click will still cause junk to fill the screen

@securedirective it's a feature that's sending keys to move the terminal cursor to the position of the mouse cursor. It needs support in whatever program is running in the terminal though which is why you might see the `^[[D` printed, we cannot detect this support though. The advice right now is to avoid alt clicking if it doesn't work in the program.

**Comment by securedirective** [comment]
@Tyriar could you clarify... Is this a feature in Linux that's doing it, and vscode is not able to see that it's happening? Or do you mean this is a feature in vscode to move the cursor, and vscode is not able to tell if the terminal shell supports it?

**Comment by Tyriar** [comment]
@securedirective it's a feature in vscode/xterm.js, you can see the PR here https://github.com/xtermjs/xterm.js/pull/896

**Comment by securedirective** [comment]
@Tyriar, what an interesting feature, and I never even knew that was there. Thank you for the perspective.

I looked through that PR and searched the settings in VSCode; unfortunately, there doesn't seem to be any way to disable that cursor-moving feature.

**Comment by notpushkin** [comment]
Yeah, that feature might be useful but the ability to disable it or, even better, change a modifier to another one (so that it doesn't conflict with the “open file in editor” one) :)

**Comment by matthias-ccri** [comment]
I'm seeing this happen regardless of the `"editor.multiCursorModifier"` setting, on Ubuntu 18.04

**Comment by Ranguna** [comment]
Just to inform that xtermjs/xterm.js/issues/3145 is waiting for a community PR for this.

**Comment by securedirective** [comment]
Um, how is `xterm.js` project related to VSCode?

**Comment by gnowland** [comment]
> Um, how is `xterm.js` project related to VSCode?

@securedirective  It's a direct dependency:
https://github.com/microsoft/vscode/blob/fb80c0e44af034df58e329e0f946a9a722ab297c/package.json#L74

**Comment by awmottaz** [comment]
I opened a PR with xterm.js which would make it possible to disable this feature: https://github.com/xtermjs/xterm.js/pull/3181

**Comment by awmottaz** [comment]
The PR was merged. What is required in order to pass the new option to xterm.js in VSCode?

**Comment by securedirective** [comment]
> 
> 
> > Um, how is `xterm.js` project related to VSCode?
> 
> Seriously? It's a direct dependency:
> 
> https://github.com/microsoft/vscode/blob/fb80c0e44af034df58e329e0f946a9a722ab297c/package.json#L74

@gnowland I meant no offense; perhaps my "Um" conveyed an unintended tone. I was simply asking, as a non-developer user who has never looked through the VSCode codebase, but is experiencing the OP's issue... how an xterm project would be blocked by an issue in VSCode.

**Comment by gnowland** [comment]
@securedirective Ah sorry, I assumed you were familiar with the package because of your previous posts (and although it’s no excuse I will refrain from posting prior to my morning coffee in the future)... message duly modified

**Comment by christhofer** [comment]
So, does anyone know how to disable it in vscode?

**Comment by notpushkin** [comment]
@Tyriar The problem with xterm.js has been fixed. Could you kindly suggest what are our next steps here?

**Comment by pelumiojebode** [comment]
Account and transportation

**Comment by pelumiojebode** [comment]
Travel using my jet round the world
ACJ-ACH
And spending money using my bitpay checkout app

**Comment by pelumiojebode** [comment]
Setup the Account - No settlement settings for now for I’m busy and won’t be staying at a location for long or preferably find a better settlement settings and finalize the account setup for use ...including bank transactions

**Comment by Tyriar** [comment]
Sorry, I've been out for a while. We updated xterm.js in vscode yesterday, so the next steps are to expose it as a setting in vscode. Searching for a similar setting `macOptionClickForcesSelection` should show how/where to do something like this.

@meganrogge if no one picks this up from the community it would be a good change for you in debt week.

**Comment by notpushkin** [comment]
@meganrogge Thank you so much! :tada:

**Comment by meganrogge** [comment]
Once it's released in insider's, to verify, set `terminal.integrated.altClickMovesCursor:false`

**Comment by sguillia** [comment]
Hi!
Thanks for fixing this

Just to understand, may I ask:
- Why will the future setting `terminal.integrated.altClickMovesCursor` default to true?
  One year ago the problem didn't happen on a fresh install, so why not fix the regression
- Why is this issue closed but the bug still exists in the latest Insiders?

Thank you

**Comment by septatrix** [comment]
> * Why will the future setting `terminal.integrated.altClickMovesCursor` default to true?
>   One year ago the problem didn't happen on a fresh install, so why not fix the regression

As far as I understand this is a new feature in `xter.m.js`. The special characters only get displayed in certain situations (probably when the running command prevents the shell from moving the cursor). Also the default setting for multicursor is <kbd>Ctrl</kbd> so you would not use <kbd>Alt</kbd> normally unless you want exactly this new feature.

**Comment by sguillia** [comment]
I just want to open a file from within the terminal out of the box

![Capture du 2021-01-18 09-48-21](https://user-images.githubusercontent.com/15984272/104892494-8b1cdb00-5972-11eb-9249-8852675dc6cf.png)

I typically do when running `tsc` in watch mode

On Linux the shortcut is Ctrl+Alt+Click even if the screenshot says otherwise

Can users do this without having to find this issue and the corresponding setting?

Also, when the `^[[D^[[D^[[D^[[D` filling happens, often the terminal gets stuck, I cannot kill the process with `Ctrl+C` anymore, should I open another issue about this?

**Comment by gnowland** [comment]
Thank you for voicing your concerns @sguillia - I agree that OFF should be the default behavior as Alt-Click is mapped to the file open command in VS Code, and ON is therefore the most likely undesired behavior.

**Comment by septatrix** [comment]
<kbd>Alt</kbd>+<kbd>Click</kbd> is not the default. The default for opening links and definitions is the one key not used by the multicursor modifier which is by default <kbd>Alt</kbd>. You can confirm this by looking in the docs https://code.visualstudio.com/docs/editor/codebasics#_multicursor-modifier or viewing the default value in the settings UI.

**Comment by sguillia** [comment]
Just tried eaa959d in Portable mode on Linux with no settings nor extension installed. The problem still reproduces.

`terminal.integrated.altClickMovesCursor` fixes the problem but I'm not sure how new users would know about this

**Comment by securedirective** [comment]
@septatrix You may be misunderstanding what this option attempts to solve. It's unfortunate that the name `altClickMovesCursor` was chosen for the setting, since the setting is not about `Alt` vs. `Ctrl`, but rather about enabling or disabling the pseudo-mouse-cursor-moving feature in xTerm entirely.

I concur with @sguillia... go back a year and VSCode didn't have this annoying broken "feature" of the terminal. Now that we've recognized it's broken and updated the xTerm dependency to a version that allows us to get rid of the broken "feature", the default behavior from a VSCode perspective should be to have the feature disabled, since we really never wanted it in the first place.

Also, as @sguillia  pointed out, when the typical user hits this strange behavior, are we just expecting them to "VSCode terminal fills with ^[[D^[[D^[[D^[[D^[[D^[[D and locks up" into Google and hope for the best?

This "feature" does not work 100%. It should be considered beta/experimental and disabled by default.

**Comment by securedirective** [comment]
@meganrogge, I can attempt a PR that
* Changes the setting name to perhaps `enableXTermCursorMove` (something that doesn't mention `Alt` vs. `Ctrl`)
* Default value of `false`

But I'm not sure how the open-source community handles requests like this. Any advice on how to proceed? Or is this a closed/merged PR and conversation over?

**Comment by Tyriar** [comment]
After re-reading this I realized this is capturing the bug (alt+click conflict) as well as the feature request (new setting), I thought they were separate issues. @meganrogge to fix let's:

- Override the value of `altClickMovesCursor ` to be false when `"editor.multiCursorModifier": "ctrlCmd"`.
- Document this override in `altClickMovesCursor`'s description, that also makes the name correct as this feature only ever works on alt+click. I think that name is good as it aligns with `macOptionClickForcesSelection`.

@securedirective we're a little tight on time as we're closing in on the end of the iteration, @meganrogge can handle it 🙂

**Comment by securedirective** [comment]
Thanks, @Tyriar and @meganrogge! I'm really looking forward to being able to click links in the terminal again.
